### ISSUE TITLE
Windows CI is too slow

### ISSUE THREAD

**Issue by JasonGross** [mit-plv/fiat-crypto#1170]
Since we added a couple more AMD64 tests, the Windows build has been timing out.

Current CI Build times are:

Stage \ Version | [Windows (8.12.0, no bedrock)](https://github.com/mit-plv/fiat-crypto/runs/5656129903?check_suite_focus=true) | [MacOS (8.14.0, with bedrock)](https://github.com/mit-plv/fiat-crypto/runs/5656130024?check_suite_focus=true) | [Ubuntu LTS (8.11.0, no bedrock)](https://github.com/mit-plv/fiat-crypto/runs/5652949675?check_suite_focus=true) | [Ubuntu (master, with bedrock)](https://github.com/mit-plv/fiat-crypto/runs/5652949651?check_suite_focus=true)
-|-|-|-|-
Install OCaml | 2m45s | 2m58s | - | -
Install Coq | 15m10s | 9m53s | 1m28s | 1m27s
Build deps | 17m56s | 24m32s | 9m3s | 20m27s
Build all | 3h9m48s | 1h52m30s | 1h25m47s | 1h35m10s
Test AMD 64 files | > 2h13m44s | 1h17m9s | 1h1m27s | 1h2m58s
Build Haskell binaries | - | - | 22m16s | 26m7s
validate | - | 1h37m20s | 48m33s | 1h22m48s
**Total** | > 6h | 5h26m27s | 3h49m48s | 4h50m44s

Part of why Windows is so much slower is that we need to build single-threaded to not run out of RAM.  I guess maybe I can multithread the AMD64 tests?  But we should maybe think about eliminating some of the slower AMD64 tests?  Or performance-optimizing the equivalence checker?  cc @dderjoel @andres-erbsen

**Comment by dderjoel** [comment]
1. Could we put multiple hint-files per invocation, such that the binary doesn't need to start from scratch synthesizing the IR that the amd file need to match? 

1. I think the files that were there before the last couple extensions were just a random selection. We could definitely scrap some of them, or merge them such that all the constructs we support are in one (fabricated) file.

I tend to do both actually, I don't know how much work the first is and if it'd be worth it. (It'd would help my project a bit too. I could then check all emitted files at the end once, rather than checking every time I write a file, but that is super-low priority).

And finally, testing in parallel makes sense. Is -j2 already enough ?

**Comment by JasonGross** [comment]
> Is -j2 already enough ?

[According to wmic, the Windows machines have only 2 cores](https://github.com/mit-plv/fiat-crypto/runs/5656129903?check_suite_focus=true#step:7:58), so anything higher than -j2 probably won't help.



> 1. Could we put multiple hint-files per invocation, such that the binary doesn't need to start from scratch synthesizing the IR that the amd file need to match?

In theory, yes.  The main difficulty is just figuring out how to set up the CLI, pipe through the file names so errors say what file the error came from, and add a mode that is strictly checking and doesn't emit the assembly as output.

**Comment by dderjoel** [comment]
I wondered, if `-j2` would do anything at all. But I just had another look into the code, the generated `Makefile` and yes should help.

> The main difficulty is just figuring out how to set up the CLI.
> and add a mode that is strictly checking and doesn't emit the assembly as output.

I am not sure what the asm-output is for (I don't use it), So I'd be fine with it being disabled, esp. on multiple files.
Also, I'd be fine with only saying 'some were invalid' and exit with `!=0`

sort of like this:
```
if( multiple files )
disable_asm_output();

for each file: 
 if checkFile(file) == invalid; exit(invalid);
```

And to debug, one can then narrow down themself.
